### ISSUE TITLE
Poseidon2 Circuit

### ISSUE THREAD

**Issue by alxiong** [EspressoSystems/jellyfish#710]
This task implements the poseidon2 permutation in plonk circuit. We may consider implementing circuits for P2-derived symmetric primitives

**Comment by MuhtasimTanmoy** [comment]
Still relevant?

**Comment by alxiong** [comment]
yes @MuhtasimTanmoy, but it's a low-priority task for us. Is this a feature request from your team? How urgent do you need it?
